src/main/java/org/sonar/plugins/checkstyle/metadata/CheckUtil.java

```diff
@@ -60,11 +62,19 @@ else if (AbstractCheck.class.isAssignableFrom(checkResult.getClass())) {
     }
 
     private static AbstractCheck getCheck(String checkName) {
-        final ClassLoader classLoader = CheckstyleMetadata.class.getClassLoader();
+        if (packageObjectFactory == null) {
+            try {
+                final ClassLoader classLoader = CheckstyleMetadata.class.getClassLoader();
+                final Set<String> packageNames = PackageNamesLoader.getPackageNames(classLoader);
+                packageObjectFactory = new PackageObjectFactory(packageNames, classLoader);
+            }
+            catch (CheckstyleException ex) {
+                throw new IllegalStateException(ex);
```

Comment:
During wrapping of exception we should provide context of execution in this case it is checkname

Reply:
There is no extra "context" to provide.

The only parameter of the method, `checkName`, is not used in this try/catch so it has no basis on the exception. The only thing to know is what method did the call and caused the exception, and that is already part of the stack trace.

Any exception that does occur, all information on the issue will be from the original exception.

Reply:
IllegalStateException("exception happened during loading of " + checkName, ex);

Reply:
Done